inst/pages/98_exercises.qmd

```diff
@@ -1152,6 +1155,64 @@ Useful functions: scater::runMDS, runNMDS, transformAssay, ggplot, scater::plotR
 
 See the OMA book for examples.
 
+**Extra** Cool Exercise : Identify Patterns and Clusters
+To make the exercise more engaging, let's do an exploratory data analysis task:
+
+#### 1. Cluster Analysis on PCoA/PCA Results:
+
+i. Perform clustering on the ordination results (e.g., k-means clustering).
+ii. Visualize the clusters on the ordination plots.
+
+```{r}
+# Setting seed for reproducibility
+set.seed(123)
+
+# Perform k-means clustering on the reduced dimensions of pcoa_bray
+pcoa_bray_clusters <- kmeans(reducedDim(pcoa_bray), centers = 3)$cluster
+
+# Plot the reduced dimensions of pcoa_bray with clusters as colors
+plotReducedDim(pcoa_bray, colour_by = pcoa_bray_clusters)
+
+# Perform k-means clustering on the reduced dimensions of pca_aitchison
+pca_aitchison_clusters <- kmeans(reducedDim(pca_aitchison), centers = 3)$cluster
+
+# Plot the reduced dimensions of pca_aitchison with clusters as colors
+plotReducedDim(pca_aitchison, colour_by = pca_aitchison_clusters)
+```
+Where : 
+'reducedDim' is a function that extracts the reduced dimensionality representation of your data (e.g., from a PCA or 
+PCoA object).
+'plotReducedDim' is a function that can plot these reduced dimensions and has a 'colour_by argument' that accepts a vector of cluster assignments.
+
+#### 2. Heatmap with Annotations:
+
+* Annotate the heatmap with additional metadata (e.g., subject groups or sample types).
+
+```{r}
+# Load necessary libraries
+library(SummarizedExperiment)
+library(pheatmap)
+
+# Example data creation
+# Assuming enterotype is a SummarizedExperiment object with sample metadata
+# and assay_data is a matrix containing assay data
+
+# Create example SummarizedExperiment object
+example_data <- matrix(runif(200), nrow = 20, ncol = 10) # Example assay data
+col_data <- DataFrame(Sample = paste0("Sample", 1:10), Group = rep(c("A", "B"), each = 5)) # Example sample metadata
+enterotype <- SummarizedExperiment(assays = list(counts = example_data), colData = col_data)
+
+# Extract sample metadata
+sample_metadata <- colData(enterotype)
+
+# Extract assay data
+assay_data <- assay(enterotype, "counts")
+
+# Plot heatmap with clustering and sample annotations
+pheatmap(assay_data, cluster_rows = TRUE, cluster_cols = TRUE, annotation_col = as.data.frame(sample_metadata))
```

Comment:
You could use sechm package which creates heatmaps directly from TreeSummarizedExperiment